### PR TITLE
[quant] Add nnq.Identity

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23469 [quant] Add nnq.Identity**

we need nnq.Identity because after fuse modules, we swap some modules as Identity.

Differential Revision: [D16530202](https://our.internmc.facebook.com/intern/diff/D16530202/)

### CODE REVIEW

File: torch/quantization/quantize.py [hunk 1]
```diff
@@ -227,6 +227,7 @@ def quantize_qat(model, run_fn, run_args, qconfig_dict=None):
     torch.nn.Conv2d: nnq.Conv2d,
     QuantStub: nnq.Quantize,
     DeQuantStub: nnq.DeQuantize,
+    nn.Identity: nnq.Identity,
```
Comment (raghuramank100) [inline torch/quantization/quantize.py]: Do we need this? We need not add nn.Identity to the list of modules to swap. nn.Identity already works for quantized tensors.

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Not sure if this change is needed, please see comments.

**Comment by jerryzh168** [comment]
We need from_float

File: torch/quantization/quantize.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/quantize.py]: oh, reason is we need to remove observer

Comment (jerryzh168) [reply torch/quantization/quantize.py]: maybe avoid adding observer for Identity?

Comment (raghuramank100) [reply torch/quantization/quantize.py]: Yes, I would recommend that. We need a list of modules for which we will not insert observers eventually.

**Comment by jerryzh168** [comment]
see https://github.com/pytorch/pytorch/pull/23500

**Pull request closed without merging**